### DISCUSSION TITLE
UNMANIC

### DISCUSSION THREAD

**Discussion by ptichalouf** [tteck/Proxmox#1976] [Ideas]
Hello thanks for your hard your but this is possible to add unmanic with IGPU ? thanks again :)

**Comment by tteck** [comment]
Wouldn't Unmanic need to be installed on an existing LXC?

**Reply by ptichalouf** [reply]
Yeah i manage to install everything with the docker script and docker compose after that but for newbies that would be great to have a 1  copy past install like tdarr :)

**Reply by tteck** [reply]
If I create a script, would you be able to test?

**Reply by ptichalouf** [reply]
Ofc :)

**Reply by tteck** [reply]
Ready for testing.

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/unmanic.sh)"
```

**Reply by tteck** [reply]
Run within the Proxmox shell.

**Reply by tteck** [reply]
If hardware acceleration is needed. Also run within the Proxmox shell.

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/hw-acceleration.sh)"
```

**Reply by ptichalouf** [reply]
so i try without adding hw for now and nothing goes to the pending list, this is better to run directly itself like tdarr and not in an other lxc i think.

**Reply by ptichalouf** [reply]
i have this error:

`2023-10-29T14:48:34:ERROR:Unmanic.PluginExecutor - [FORMATTED] - Exception while carrying out 'library_management.file_test' plugin runner 'video_transcoder2'

Traceback (most recent call last):

  File "/usr/local/lib/python3.11/dist-packages/unmanic/libs/unplugins/executor.py", line 256, in execute_plugin_runner

    runner(data)

  File "/root/.unmanic/plugins/video_transcoder2/plugin.py", line 175, in on_library_management_file_test

    probe = Probe.init_probe(data, logger, allowed_mimetypes=['video'])

            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/root/.unmanic/plugins/video_transcoder2/lib/ffmpeg/probe.py", line 169, in init_probe

    probe = Probe(logger, allowed_mimetypes=allowed_mimetypes)

            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/root/.unmanic/plugins/video_transcoder2/lib/ffmpeg/probe.py", line 117, in __init__

    raise Exception("Unable to find executable 'ffprobe'. Please ensure that FFmpeg is installed correctly.")

Exception: Unable to find executable 'ffprobe'. Please ensure that FFmpeg is installed correctly.`

**Reply by tteck** [reply]
If in a separate LXC, it looks like local files only. I can add FFmpeg, you would think that installing a "plugin" it would add necessary dependencies.

**Reply by ptichalouf** [reply]
i think that not a big deal peoples will make mount point. i make this personnaly of maybe you can add a choice before installing anything.  And i think you need to add ffmpeg. i see some error on tdarr too but we could talk about that later if you want

**Reply by tteck** [reply]
And the ability to link remote installations

**Reply by tteck** [reply]
`apt-get install ffmpeg`

**Reply by ptichalouf** [reply]
link remote works great :)

**Reply by ptichalouf** [reply]
replied below sorry

**Comment by ptichalouf** [comment]
> `apt-get install ffmpeg`
so with this pending list is no more empty. i try to add HW with the other script but still can't trancode

`LOG:

Input #0, matroska,webm, from '/mnt/data/media/tv/Brooklyn Nine-Nine (2013) [tvdbid-269586]/Season 02/Brooklyn Nine-Nine (2013) - S02E06 - Jake and Sophia [WEBDL-1080p][AC3 2.0][h264]-LiBERTY.mkv':

   Metadata:

     title : Brooklyn.Nine-Nine.S02E06.FRENCH.1080p.WEB-DL.H264-LiBERTY

     encoder : libebml v1.3.0 + libmatroska v1.4.1

     creation_time : 2015-07-06T11:35:27.000000Z

   Duration: 00:20:45.82, start: 0.000000, bitrate: 5346 kb/s

   Stream #0:0(fre): Video: h264 (High), yuv420p(tv, bt709, progressive), 1920x1080 [SAR 1:1 DAR 16:9], 25 fps, 25 tbr, 1k tbn (default)

     Metadata:

       title : Brooklyn.Nine-Nine.S02E06.FRENCH.1080p.WEB-DL.H264-LiBERTY

   Stream #0:1(fre): Audio: ac3, 48000 Hz, stereo, fltp, 192 kb/s (default)

     Metadata:

       title : Brooklyn.Nine-Nine.S02E06.FRENCH.1080p.WEB-DL.H264-LiBERTY

Stream mapping:

   Stream #0:0 (h264) -> hwupload:default

   format:default -> Stream #0:0 (hevc_qsv)

   Stream #0:1 -> #0:1 (copy)

Press [q] to stop, [?] for help

[AVHWFramesContext @ 0x55cb0c811280] Failed to sync surface 0xc876110: 6 (invalid VASurfaceID).

[hwupload @ 0x55cb0c380740] Failed to upload frame: -5.

Error while filtering: Input/output error

Failed to inject frame into filter network: Input/output error

Error while processing the decoded data for stream #0:0`

**Reply by tteck** [reply]
Don't know what any of that means. 🤷🏻‍♂️ 

My primary focus revolves around the development of application LXCs, and I'm not specialized in providing support for operational issues that may arise with these applications.

**Reply by akmarwah03** [reply]
@ptichalouf , how were you able to solve this, I am trying with qsv and getting same error

**Reply by ptichalouf** [reply]
@akmarwah03 using docker lxc with portainer and installing unmanic via portainer with this template : https://github.com/Lissy93/portainer-templates

**Comment by tteck** [comment]
> i try to add HW with the other script but still can't trancode

Privileged container?

**Reply by ptichalouf** [reply]
yes i try stuff and i will reply here if i find something

**Comment by di5cord20** [comment]
> Ready for testing.
> 
> ```
> bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/unmanic.sh)"
> ```
Works perfectly.  Ran in Proxmox shell selecting the Tdarr LXC previously created (which was configured privileged).  As @ptichalouf mentioned earlier, ideally this would be it's own LXC (similar to how Tdarr LXC script works).  If that we available, I'll kill my Tdarr LXC (which is now my Unmanic LXC labelled tdarr) and start fresh.

**Reply by ptichalouf** [reply]
i will try in tdarr lxc so :)

**Comment by tteck** [comment]
New test, Its own LXC

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/unmanic.sh)"
```

**Reply by tteck** [reply]
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/unmanic.sh)"
```
## Will be deleted shortly!

**Reply by ptichalouf** [reply]
i add this error on the script 

`   __  __                            _
  / / / /___  ____ ___  ____ _____  (_)____
 / / / / __ \/ __ `__ \/ __ `/ __ \/ / ___/
/ /_/ / / / / / / / / / /_/ / / / / / /__
\____/_/ /_/_/ /_/ /_/\__,_/_/ /_/_/\___/

Using Default Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 0
Using Root Password: Automatic Login
Using Container ID: 121
Using Hostname: unmanic
Using Disk Size: 4GB
Allocated Cores 2
Allocated Ram 2048
Using Bridge: vmbr0
Using Static IP Address: dhcp
Using Gateway Address: Default
Disable IPv6: No
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server Address: Host
Using MAC Address: Default
Using VLAN Tag: Default
Enable Root SSH Access: No
Enable Verbose Mode: No
Creating a Unmanic LXC using the above default settings
 ✓ Using local for Template Storage.
 ✓ Using UNRAID_Container for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 121 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.1.248 2a01:cb08:a20:5600:608b:4bff:fee5:501a 
 ✓ Internet Connected
 ✓ DNS Resolved github.com to 140.82.121.4
 ✓ Updated Container OS
 ✓ Installed Dependencies
 - Setting Up Hardware Acceleration.../bin/chgrp: cannot access '/dev/dri': No such file or directory

[ERROR] in line 31: exit code 1: while executing command /bin/chgrp video /dev/dri
root@node1:~# `

**Reply by ptichalouf** [reply]
if u fix it could you leave the script until tommorow :) i going to sleep :)

**Reply by tteck** [reply]
deleting the first script, not the one that creates its own LXC.

**Reply by ptichalouf** [reply]
but any idea why i have this error ? i58600T and /dev/dri works great anywhere else

**Reply by di5cord20** [reply]
I got the same error as @ptichalouf  when running the new script ```bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/unmanic.sh)"```

**Reply by tteck** [reply]
fixed

**Reply by di5cord20** [reply]
Worked perfectly!  Thanks so much for making this script.  

For anyone planning to use this script, I'd suggest increasing the disk size (especially if used for encoding as you'll run out of space quickly).  Similary, if you're using CPU for encoding, you'll want to increase the cores from the default (if you have them available).  For reference, I'm using an N100 with 4 CPUs and have allocated them all to the container.  While encoding 1 H264 to 265, my cpu maxed out at ~70%.

Lastly, there are a lot of dependencies during the install.  Just be patient and let the script complete.

**Reply by ptichalouf** [reply]
Already 3 nodes with the new script i made a mount point everything is gorgeous ! Thanks again for your time and work
